src/pool/srv_pool.c

```diff
@@ -4147,6 +4157,9 @@ static int ds_pool_mark_upgrade_completed(uuid_t pool_uuid,
 				"of pool, %d.\n", DP_UUID(pool_uuid), rc1);
 			D_GOTO(out_tx, rc1);
 		}
+		/* also bump cached version */
+		svc->ps_global_version = DS_POOL_GLOBAL_VERSION;
```

Comment:
Strictly speaking, we should bump ps_global_version only if tx commits successfully.